File: 20242Q/poo/laboratorio/tp04/_soluciones/ej4/Ellipse.java
package ar.edu.itba.poo.soltp4.ej4;

public class Ellipse extends Figure {

    protected final Point centerPoint;
    protected final double sMayorAxis, sMinorAxis;

    public Ellipse(Point centerPoint, double sMayorAxis, double sMinorAxis) {
        this.centerPoint = centerPoint;
        this.sMayorAxis = sMayorAxis;
        this.sMinorAxis = sMinorAxis;
    }

    @Override
    public double area() {
        return Math.PI / 4 * sMayorAxis * sMinorAxis;
    }

    @Override
    public double perimeter() {
        return Math.PI / 2 * (sMayorAxis + sMinorAxis);
    }

    @Override
    public String toString() {
        return String.format("Elipse [Centro: %s, DMayor: %.2f, DMenor: %.2f]", centerPoint, sMayorAxis, sMinorAxis);
    }

}

File: 20242Q/poo/laboratorio/tp08/_soluciones/ej4/ellipse.rb
require_relative 'figure'

class Ellipse < Figure
  def initialize(center_point, smayor_axis, sminor_axis)
    @center_point = center_point
    @smayor_axis = smayor_axis
    @sminor_axis = sminor_axis
  end

  def area
    Math::PI / 4 * @smayor_axis * @sminor_axis
  end

  def perimeter
    Math::PI / 2 * (@smayor_axis + @sminor_axis)
  end

  def to_s
    "Elipse [Centro: #{@center_point}, DMayor: #{@smayor_axis}, DMenor: #{@sminor_axis}]"
  end
end
